transport/src/main/java/io/netty/channel/socket/nio/NioChannelOption.java

```diff
@@ -19,33 +19,38 @@
 import io.netty.channel.ChannelOption;
 
 import java.io.IOException;
-import java.net.SocketOption;
-import java.nio.channels.NetworkChannel;
+import java.nio.channels.Channel;
 import java.util.Set;
 
 /**
- * Provides {@link ChannelOption} over a given {@link SocketOption} which is then passed through the underlying
- * {@link NetworkChannel}.
+ * Provides {@link ChannelOption} over a given {@link java.net.SocketOption} which is then passed through the underlying
+ * {@link java.nio.channels.NetworkChannel}.
  */
 public final class NioChannelOption<T> extends ChannelOption<T> {
 
-    private final SocketOption<T> option;
+    private final java.net.SocketOption<T> option;
 
     @SuppressWarnings("deprecation")
-    private NioChannelOption(SocketOption<T> option) {
+    private NioChannelOption(java.net.SocketOption<T> option) {
         super(option.name());
         this.option = option;
     }
 
     /**
-     * Returns a {@link ChannelOption} for the given {@link SocketOption}.
+     * Returns a {@link ChannelOption} for the given {@link java.net.SocketOption}.
      */
-    public static <T> ChannelOption<T> of(SocketOption<T> option) {
+    public static <T> ChannelOption<T> of(java.net.SocketOption<T> option) {
         return new NioChannelOption<T>(option);
     }
 
+    // It's important to not use java.nio.channels.NetworkChannel in the method signature directly as this will lead
```

Comment:
This comment feels weak or misleading, as it's actually the callers of this method that cause the trouble. Maybe:
> It's important to not use java.nio.channels.NetworkChannel as otherwise the classes that sometimes call this method may not be used on Java 6, as method linking can happen eagerly even if this method was not actually called at runtime.